battle.go

```diff
@@ -481,28 +484,53 @@ type BattleContext struct {
 	Targets   []target // An array of all possible targets that the Pokemon can act on
 }
 
-// Gets all the active Pokemon (targets) in the battle
+// Gets a deep copy of all the active Pokemon (targets) in the battle
 func (b *Battle) GetTargets() []target {
 	targets := make([]target, 0)
 	for partyID, party := range b.parties {
 		for slot, active := range party.activePokemon {
+			var pkmn Pokemon
+			bytes, _ := json.Marshal(active)
+			json.Unmarshal(bytes, &pkmn)
```

Comment:
This seems like a gimmicky way of making a deep copy. Why can't we use a golang lib function?

Reply:
The sort answer is no, [Go does not have a standard library function to deep copy an object](https://stackoverflow.com/questions/56355212/deep-copying-data-structures-in-golang), for some reason. I saw somebody else's benchmarks, and this was the fastest.